crates/test-utils/src/deploy_helpers.rs

```diff
@@ -634,8 +634,21 @@ impl DeployedKakarot {
     }
 }
 
+/// Returns the dumped Katana state with deployed Kakarot + EVM contracts.
+pub fn dumped_kakarot_state() -> SerializableState {
+    // Get root path
+    let dir = root_project_path!(".katana/dump.json");
+
+    // Create SerializableState from dumped state
+    let state = std::fs::read_to_string(dir).expect("Failed to read Katana dump");
+    serde_json::from_str(&state).expect("Failed to deserialize Katana dump")
+}
+
 /// Returns a `StarknetConfig` instance customized for Kakarot.
-pub fn kakarot_starknet_config() -> StarknetConfig {
+/// If `with_dumped_state` is true, the config will be initialized with the dumped state.
+pub fn kakarot_starknet_config(with_dumped_state: bool) -> StarknetConfig {
+    let state = if with_dumped_state { Some(dumped_kakarot_state()) } else { None };
```

Comment:
why not `init_state`?

Reply:
wdym?
but to be fair, maybe this kakarot_starknet_config should be a method, implemented on the struct.
let's open an issue to refactor it?